zebra-state/src/service/finalized_state/zebra_db/block/tests/snapshot.rs

```diff
@@ -0,0 +1,281 @@
+//! Database snapshot tests for blocks and transactions.
+//!
+//! These tests check the values returned by all the APIs in the `zebra_db::block` module,
+//! iterating through the possible arguments as needed:
+//! - tip
+//! - hash(height: tip..=0)
+//! - height(hash: heights)
+//! - block(height: tip..=0)
+//! - transaction(hash: blocks.transactions.hashes)
+//!
+//! But skips the following trivial variations:
+//! - is_empty: covered by `tip == None`
+//! - block(hash): covered by `height(hash) and block(height)`
+//!
+//! These tests use fixed test vectors, based on the results of other database queries.
+//!
+//! # Fixing Test Failures
+//!
+//! If this test fails, run `cargo insta review` to update the test snapshots,
+//! then commit the `test_*.snap` files using git.
+//!
+//! # TODO
+//!
+//! Test shielded data, and data activated in Overwinter and later network upgrades.
+
+use std::sync::Arc;
+
+use serde::{Deserialize, Serialize};
+
+use zebra_chain::{
+    block::{self, Block, Height},
+    parameters::Network::{self, *},
+    serialization::{ZcashDeserializeInto, ZcashSerialize},
+    transaction::Transaction,
+};
+
+use crate::{
+    service::finalized_state::{disk_format::TransactionLocation, FinalizedState},
+    Config,
+};
+
+/// Tip structure for RON snapshots.
```

Comment:
I think it can be good to make a short description of what RON is. The description can be in the doc header so the reader will know what are we referring to.

For example, are we talking about https://github.com/ron-rs/ron or https://replicated.cc/ ?

Reply:
I'll do this in the next PR, since this one is running the full sync tests right now.

Reply:
Feel free to resolve the conversation as needed.

Reply:
I explained RON in [f706967](https://github.com/ZcashFoundation/zebra/pull/3717/commits/f706967df446543dc1e4d8c5d2fe423b983a10aa)